File: 2007-Pycon-Paris/ex_exception/exemple_corrige.py
from sys import exit

def main():
    try:
        tampon = raw_input("Entrez votre nom : ")
    except EOFError:
        exit(0)

    try:
        if tampon[len(tampon)-1] == '\n':
            tampon = tampon[:len(tampon)-1]
    except IndexError:
        pass

    print "Bonjour %s !" % tampon
    exit(0)

if __name__ == "__main__":
    main()

File: 2007-Pycon-Paris/ex_exception/exemple_corrige.c
#include <stdlib.h>
#include <stdio.h>
#include <string.h>

#define TAILLE 10

int main()
{
    char *tampon, *ok;
    size_t len;

    tampon = (char *)malloc(TAILLE);
    if (!tampon) {
        fprintf(stderr, "Impossible d'allouer le tampon !\n");
        exit(EXIT_FAILURE);
    }

    printf("Entrez votre nom : ");
    fflush(stdout);

    ok = fgets(tampon, TAILLE, stdin);
    if (!ok) {
        free(tampon);
        fprintf(stderr, "Erreur lors de la lecture du nom !\n");
        exit(EXIT_FAILURE);
    }

    len = strlen(tampon);
    if (1 <= len && tampon[len-1] == '\n') {
        tampon[len-1] = '\0';
    }

    printf("Bonjour %s !\n", tampon);
    free(tampon);
    exit(EXIT_SUCCESS);
}
